ts/components/FiscalCodeLandscapeOverlay.tsx

```diff
@@ -52,93 +53,130 @@ const styles = StyleSheet.create({
-    BackHandler.removeEventListener("hardwareBackPress", this.handleBackPress);
-    // if there is an active timeout, clear it!
-    if (this.scrollTimeout !== undefined) {
-      clearTimeout(this.scrollTimeout);
-      // eslint-disable-next-line
-      this.scrollTimeout = undefined;
-    }
-  }
+  const handleBackPress = () => {
+    // On backpress the component gets unmounted, so the brightness is restored by the
+    // cleanup function
+    props.onCancel();
+    return true;
+  };
 
-  private scrollToEnd = () => {
-    if (this.props.showBackSide && this.ScrollVewRef.current) {
+  const scrollToEnd = () => {
+    if (props.showBackSide && ScrollViewRef.current) {
       // dalay the scroll to end command to wait until the ingress animation is completed
       // eslint-disable-next-line
-      this.scrollTimeout = setTimeout(() => {
-        if (this.ScrollVewRef.current) {
-          this.ScrollVewRef.current.scrollToEnd({ animated: true });
+      scrollTimeout = setTimeout(() => {
+        if (ScrollViewRef.current) {
+          ScrollViewRef.current.scrollToEnd({ animated: true });
         }
       }, 300);
     }
   };
 
-  public render() {
-    return (
-      <Container style={{ backgroundColor: customVariables.brandDarkGray }}>
-        <AppHeader noLeft={true} dark={true}>
-          <Body />
-        </AppHeader>
-        <StatusBar
-          backgroundColor={customVariables.brandDarkGray}
-          barStyle={"light-content"}
-        />
-        <ScrollView
-          style={styles.content}
-          ref={this.ScrollVewRef}
-          onLayout={this.scrollToEnd}
-        >
-          <View style={styles.headerSpacer} />
-          <View spacer={true} />
-          <View>
-            <FiscalCodeComponent
-              type={"Landscape"}
-              profile={this.props.profile}
-              getBackSide={false}
-              municipality={this.props.municipality}
-            />
-          </View>
-
-          <View spacer={true} />
+  React.useEffect(() => {
+    BackHandler.addEventListener("hardwareBackPress", handleBackPress);
+    return () => {
+      BackHandler.removeEventListener("hardwareBackPress", handleBackPress);
+      // if there is an active timeout, clear it!
+      if (scrollTimeout !== undefined) {
+        clearTimeout(scrollTimeout);
+        // eslint-disable-next-line
+        scrollTimeout = undefined;
+      }
+    };
+  }, []);
+
+  // Brightness effect manager
+  React.useEffect(() => {
+    const myBrightF = async () =>
+      await getBrightness()
+        .fold(
+          () => 0.5, // If fails, resort to a default value
+          _ => _
+        )
+        .run();
+
+    const myBright = myBrightF();
 
+    const mySetBrightF = async () => await setBrightness(HIGH_BRIGHTNESS).run();
+
+    void mySetBrightF();
+
+    return () => {
+      const restoreDeviceBrightnessF = async () => {
+        const deviceB = await myBright;
+        if (deviceB) {
```

Comment:
@Undermaken In my opinion, using a state variable can't work if the effect has a void skipping array as the second argument. In the last line, the proposed code restores the brightness always to `0.1` not to the value saved in `defaultBrightness`. Maybe I'm wrong, but in my limited understanding of React lifecycle, this is due to the fact that the function `setDefaultBrightness` is not synchronous and schedules the state update _after_ the rendering... but when the effect skip array is `[]` the effect gets never executed again, hence it can never see the new state. As a result, we must assume that the effect is executed only once, and it always uses the initial state (this is true also for the cleanup function, which sees the _current_ state not the next)
I've read a similar argument in the docs: _If you pass an empty array ([]), the props and state inside the effect will always have their initial values._   https://reactjs.org/docs/hooks-effect.html . On the other hand, I agree on returning `undefined` if you can't read the brightness.

Reply:
@GiovanniMancini 
> In my opinion, using a state variable can't work if the effect has a void skipping array as the second argument
good point, you are right.
This code will be executed once so storing a value in a React state has no effect

Thank you ;)

So I guess considering the `undefined` case should be more safe, also when we set the brightness to the maximum (if we don't know nothing about current brightness we should avoid to set it to the maximum since we can't restore it back)